seqfu/1.20.3/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Set global variables
+ARG SEQFU_VER="1.20.3"
+
+# Stage 1: Build Dockerfile
+FROM ubuntu:focal AS builder
+ARG SEQFU_VER
+
+# Metadata
+LABEL base.image="ubuntu:focal" \
+      dockerfile.version="1" \
+      software="SeqFu" \
+      software.version="${SEQFU_VER}" \
+      description="A general-purpose program to manipulate and parse information from FASTA/FASTQ files, supporting gzipped input files." \
+      website="https://github.com/telatin/seqfu2" \
+      license.url="https://github.com/telatin/seqfu2?tab=GPL-3.0-1-ov-file#readme" \
+      maintainer="Taylor K. Paisie" \
+      maintainer.email="ltj8@cdc.gov"
+
+# Set non-interactive frontend and update PATH
+ENV DEBIAN_FRONTEND=noninteractive \
+    PATH="/root/.nimble/bin:${PATH}"
+
+# Install required dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    wget \
+    unzip \
+    build-essential \
+    zlib1g-dev \
+    curl \
+    python3 \
+    python3-pip \
+    git \
+    ca-certificates && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set the working directory
+WORKDIR /root
+
+# Install Nim and SeqFu
+RUN curl https://nim-lang.org/choosenim/init.sh -sSf | sh -s -- -y --ver 1.22.2 && \
+    wget https://github.com/telatin/seqfu2/archive/refs/tags/v${SEQFU_VER}.tar.gz && \
+    tar -xzf v${SEQFU_VER}.tar.gz && \
+    rm v${SEQFU_VER}.tar.gz && \
+    cd seqfu2-${SEQFU_VER} && \
+    make
+
+# Copy the built binary to a location in the PATH
+RUN ls -l /root/seqfu2-${SEQFU_VER}/bin && \
+    cp -R /root/seqfu2-${SEQFU_VER}/bin/* /usr/local/bin/ && \
+    chmod +x /usr/local/bin/seqfu
+
+# Stage 2: Create the final image
+FROM ubuntu:focal AS app
+ARG SEQFU_VER
+
+# Copy the necessary files from the builder
+COPY --from=builder /usr/local/bin /usr/local/bin
+
+# Install minimal runtime dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl \
+    wget && \
+    rm -rf /var/lib/apt/lists/*
+
+# Set working directory
+WORKDIR /data
+
+# Verify installation
+RUN ls -l /usr/local/bin && \
+seqfu --version && seqfu --help
+
```

Comment:
This is looking great!

If you're looking for assistance with creating tests, seqtk uses some fastq files from this repository in [its tests](https://github.com/StaPH-B/docker-builds/blob/296e5dd955ed2af08bde35db7bdbf8b6a84b0293/seqtk/1.4/Dockerfile#L65)

I imagine for seqfu it'd be someting like

```
RUN wget -q https://github.com/StaPH-B/docker-builds/raw/master/tests/SARS-CoV-2/SRR13957123_1.fastq.gz && \
    wget -q https://github.com/StaPH-B/docker-builds/raw/master/tests/SARS-CoV-2/SRR13957123_2.fastq.gz && \
seqfu count -f SRR13957123_1.fastq.gz -r SRR13957123_2.fastq.gz
```

SeqFu also looks like it comes with its [own tests and files](https://github.com/telatin/seqfu2/blob/main/test/mini.sh) which may be useful.

Reply:
yes, so i originally had all the test files in the container but you mentioned it wasn't necessary, so maybe i'll just put in the seqfu test fastq files you suggested in the comment for the seqfu test in the dockerfile.  Also thank you for your help and suggestions!!!